### ISSUE TITLE
Port twisted.cred.checkers to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7834]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7834|
|Type|enhancement|
|Created|2015-03-26 07:08:23Z|
|Branch|https://github.com/twisted/twisted/tree/twcred-py3-7834-3|

<details><summary>Searchable metadata</summary>

```
trac-id__7834 7834
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_twcred_py3_7834_3 branches/twcred-py3-7834-3
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1427353703713011 1427353703713011
changetime__1432200916864430 1432200916864430
version__None None
owner__hawkowl hawkowl
cc__edorum
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [44293]) Branching to twcred-py3-7834.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Ready for review when [#6176](https://github.com/twisted/twisted/issues/6176) lands.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

(In [44732]) Branching to twcred-py3-7834-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl removed owner|
|-|-|

Builders spun, green looking, please review.

I cleaned up some tests as well, and added some docstrings.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Changes look good. Thanks!

Do we still need to import PAM ?

----

I find this hard to read this as it hide the actual call to login....  but the
new code is better than the old one :)

```
login = self.successResultOf(self.portal.login(
    DerivedCredentials(b"bob", b"hello"), self, ITestable))
```

maybe this is cleaner

```

deferred = self.portal.login(
    DerivedCredentials(b"bob", b"hello"), self, ITestable)

iface, impl, logout = self.successResultOf(deferred)

```


-------

Please check the PAM part and merge after the branch is updated with trunk and PAM is no longer used.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

(In [44894]) Branching to twcred-py3-7834-3.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl commented|
|-|-|

(In [44895]) Merge forward, refs [#7834](https://github.com/twisted/twisted/issues/7834)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_9"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44898]) Merge twcred-py3-7834-3: Port twisted.cred.checkers to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7834](https://github.com/twisted/twisted/issues/7834)
